Conceptpower+Spring/src/main/java/edu/asu/conceptpower/rest/ConceptSearch.java

```diff
@@ -103,24 +112,15 @@ private void initBinder(WebDataBinder binder) {
 
         for (Field field : conceptSearchParameters.getClass().getDeclaredFields()) {
             field.setAccessible(true);
-            if (field.getName().equalsIgnoreCase("type_uri")) {
-                if (field.get(conceptSearchParameters) != null) {
-                    searchFields.put("type_id",
-                            uriHelper.getTypeId(String.valueOf(field.get(conceptSearchParameters))));
-                }
-            } else if (SearchParamters.OPERATOR.equalsIgnoreCase(field.getName())) {
-                // If the value is null, then operator will be OR by default
-                if (field.get(conceptSearchParameters) != null) {
-                    operator = String.valueOf(field.get(conceptSearchParameters)).toUpperCase();
-                }
-            } else if (SearchParamters.PAGE.equalsIgnoreCase(field.getName())) {
-                page = field.get(conceptSearchParameters) != null ? (Integer) field.get(conceptSearchParameters) : page;
-            } else if (SearchParamters.NUMBER_OF_RECORDS_PER_PAGE.equalsIgnoreCase(field.getName())) {
-                numberOfRecordsPerPage = field.get(conceptSearchParameters) != null
-                        ? (Integer) field.get(conceptSearchParameters) : numberOfRecordsPerPage;
-            } else if (field.get(conceptSearchParameters) != null) {
-                searchFields.put(field.getName().trim(), String.valueOf(field.get(conceptSearchParameters)).trim());
-            }
+            searchFields.put(field.getName().trim(), String.valueOf(field.get(conceptSearchParameters)).trim());
+        }
+
+        preprocessConceptSearchParameterBean(conceptSearchParameters);
+        PropertyDescriptor pageDescriptor = new PropertyDescriptor("page", ConceptSearchParameters.class, "getPage",
```

Comment:
What I thought you would do:
In
```
} else if (field.get(conceptSearchParameters) != null) {
```
(line 122)
create property descriptor and call read method to add field value to searchFields. 
This way the getter is called which will do whatever modification needs to be done on the submitted value (e.g. removing trailing slash).